internal/functions/compare_versions.go

```diff
@@ -0,0 +1,163 @@
+package functions
```

Comment:
I think we should move the functions also in the provider package and folder.

Reply:
This was moved to the internal/provider